crates/weaver_resolver/README.md

```diff
@@ -33,14 +35,80 @@ involves the following steps:
 - Resolve iteratively all semantic conventions. This involves the maintenance
   of an unresolved semantic convention list and a resolved semantic convention
   list. The resolution process involves the following steps:
-  - Resolve iteratively all attributes `ref` until no more resolvable `ref` are
-    found.
   - Resolve iteratively all `extends` parent/child clauses until no more
     resolvable `extends` are found. The extended entity inherits prefix,
     attributes, and constraints from the parent entity.
+  - Resolve iteratively all attributes `ref` until no more resolvable `ref` are
+    found.
 - Apply constraints `any_of` and `include`.
 - Validate the resolved semantic conventions
   - No more unresolved `ref` or `extends` clauses. The unresolved list should
     be empty.
   - All constraints satisfied.
 
+## Lineage
+
+The resolution process can optionally compute the lineage for each attribute of
+a semantic convention registry. The lineage as such is not part of the syntax
+and structure of a semantic convention; rather, it's an extension produced by
+the `weaver` tool, intended for use in scenarios such as:
+- A semconv author wishes to verify the exact path followed by the resolution
+process in the case of a complex cascade of inheritance across multiple levels
+between groups.
+- A documentation process aims to add lineage information to the documentation
+of each attribute to allow readers to understand where the definitions of each
+field were defined.
+- The lineage information of a semantic convention could eventually feed into
+an enterprise data catalog to improve the data governance process.
+
+The general structure of the lineage generated by the resolution process is as
+follows:
+
+```json
+{
+  "registry_url": "<registry url>",
+  "groups": [
+    {
+      // Standard semconv fields 
+      "id": "<group id>",
+      "type": "<group type>",
+      "brief": "<group brief>",
+      "attributes": [ /* ... */ ],
+      "lineage": {
+        // `source_file` is either:
+        // - a relative path in the case of a registry that has been explicitly
```

Comment:
In the relative path case, I'm concerned about how this evolves from file->file->file in the application telemetry schema case.


A consumer of this resolution, do they preserve relative files or collapse them to some absolutely URL where they concsumed the original source?

For example:

- Semconv resolves their metrics and has lineage with relative paths.
- Application A creates their own schema using semconv, and resolves.
- When referring to lineage, do I see relative paths or urls or both?

This may be jumping the gun a bit, but if you plan to resolve this scenario using the same string field `source_file`, that's ok, I'm just concerned about how this "scales" from semconv -> application.

Reply:
I agree, this part is not fully cooked. However, I believe the only viable option for a published resolved schema is to support only URLs, because a local path will always be problematic, even if it's an absolute path. This is definitely something we need to resolve. I will add a note in the document to continue this discussion later.